### PR TITLE
fix client memory

### PR DESCRIPTION
**Pull request by QuinnWass**
Screenshot from 2020-10-23 16-04-39

Unit Tests Pass

### CODE REVIEW

**Review by Arjunica — APPROVED** [review]
thanks for catching that! Approved

**Comment by galaxyonfire** [comment]
Still not getting a response for corrected input after calling client with bad input
image

File: src/places_client.c [hunk 1]
```diff
@@ -41,25 +38,39 @@ places_server_1(char *host, char *city, char *state)
...
+		{
+			free(find_closest_airport_given_name_1_arg.city);
+		}
+		if(find_closest_airport_given_name_1_arg.state != NULL)
+		{
+			free(find_closest_airport_given_name_1_arg.state);
+		}
+		std::string s_city;
+		std::string s_state;
 		printf("Could not narrow down city and state. Please give longer name for the city and/or State\n");
```
Comment (galaxyonfire) [inline src/places_client.c:55]: maybe to make it specific to each error status, we should print the error message in each `if` to specify what needs to be clarified.

**Comment by galaxyonfire** [comment]
Error still occurs if input w/o quotes
image

**Comment by Arjunica** [comment]
> Error still occurs if input w/o quotes

I think something had to have changed b/c this all worked yesterday. so there has to be something in terms of memory getting messed up. @QuinnWass is there a reason you switched scanf() -> getline(). I feel like this file has 2 different styles {c, c++} if we wanna just us c++ I do not have an issue but I do not think our in/ouput just use both languages

**Comment by galaxyonfire** [comment]
ah, figured it out, needed to put `new york city`, i guess there is more than 1 new york entry for the NY state. Verified that it works otherwise.

File: src/places_client.c [hunk 2]
```diff
@@ -39,24 +36,41 @@ places_server_1(char *host, char *city, char *state)
...
 		if(result_1->status == -1){
-			printf("city\n");
```
Comment (galaxyonfire) [inline src/places_client.c:45]: maybe print out `Current state: $state` just so client is reminded what state they are searching in.

**Review by Arjunica — APPROVED** [review]
Aprroved. Merge into main

File: src/places_client.c (hunk 1 above)
Comment (QuinnWass) [reply src/places_client.c:55]: Sure

**Pull request merged**